File: src/com/app/test/util/ArrayUtils.java
package com.app.test.util;


/**
 * @author lcx
 * Created at 2020.1.6
 * Describe:
 */
public class ArrayUtils {
    static final String[] EMPTY_STRING_ARRAY = new String[0];


    static void reverse(Object[] array) {
        if (array != null) {
            reverse(array, 0, array.length);
        }
    }

    public static void reverse(long[] array) {
        if (array != null) {
            reverse(array, 0, array.length);
        }
    }

    public static void reverse(int[] array) {
        if (array != null) {
            reverse(array, 0, array.length);
        }
    }

    public static void reverse(short[] array) {
        if (array != null) {
            reverse(array, 0, array.length);
        }
    }

    public static void reverse(char[] array) {
        if (array != null) {
            reverse(array, 0, array.length);
        }
    }

    public static void reverse(byte[] array) {
        if (array != null) {
            reverse(array, 0, array.length);
        }
    }

    public static void reverse(double[] array) {
        if (array != null) {
            reverse(array, 0, array.length);
        }
    }

    public static void reverse(float[] array) {
        if (array != null) {
            reverse((float[]) array, 0, array.length);
        }
    }

    public static void reverse(boolean[] array) {
        if (array != null) {
            reverse(array, 0, array.length);
        }
    }

    public static void reverse(boolean[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = Math.max(startIndexInclusive, 0);

            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                boolean tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }

    public static void reverse(byte[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = Math.max(startIndexInclusive, 0);
            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                byte tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }

    public static void reverse(char[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = Math.max(startIndexInclusive, 0);
            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                char tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }

    public static void reverse(double[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = Math.max(startIndexInclusive, 0);
            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                double tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }

    public static void reverse(float[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = Math.max(startIndexInclusive, 0);
            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                float tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }

    public static void reverse(int[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = Math.max(startIndexInclusive, 0);
            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                int tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }

    public static void reverse(long[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = Math.max(startIndexInclusive, 0);
            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                long tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }

    public static void reverse(Object[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = startIndexInclusive < 0 ? 0 : startIndexInclusive;
            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                Object tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }

    public static void reverse(short[] array, int startIndexInclusive, int endIndexExclusive) {
        if (array != null) {
            int i = startIndexInclusive < 0 ? 0 : startIndexInclusive;
            for (int j = Math.min(array.length, endIndexExclusive) - 1; j > i; ++i) {
                short tmp = array[j];
                array[j] = array[i];
                array[i] = tmp;
                --j;
            }

        }
    }
}

File: src/com/app/test/util/ArrayUtils.kt
package com.app.test.util

/**
 * @author lcx
 * Created at 2020.1.6
 * Describe:
 */
object ArrayUtils {
    @JvmField
    val EMPTY_STRING_ARRAY = arrayOfNulls<String>(0)
    fun reverse(array: Array<Any?>?) {
        if (array != null) {
            reverse(array, 0, array.size)
        }
    }

    fun reverse(array: LongArray?) {
        if (array != null) {
            reverse(array, 0, array.size)
        }
    }

    fun reverse(array: IntArray?) {
        if (array != null) {
            reverse(array, 0, array.size)
        }
    }

    fun reverse(array: ShortArray?) {
        if (array != null) {
            reverse(array, 0, array.size)
        }
    }

    fun reverse(array: CharArray?) {
        if (array != null) {
            reverse(array, 0, array.size)
        }
    }

    fun reverse(array: ByteArray?) {
        if (array != null) {
            reverse(array, 0, array.size)
        }
    }

    fun reverse(array: DoubleArray?) {
        if (array != null) {
            reverse(array, 0, array.size)
        }
    }

    fun reverse(array: FloatArray?) {
        if (array != null) {
            reverse(array as FloatArray?, 0, array.size)
        }
    }

    fun reverse(array: BooleanArray?) {
        if (array != null) {
            reverse(array, 0, array.size)
        }
    }

    fun reverse(array: BooleanArray?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = Math.max(startIndexInclusive, 0)
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }

    fun reverse(array: ByteArray?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = Math.max(startIndexInclusive, 0)
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }

    fun reverse(array: CharArray?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = Math.max(startIndexInclusive, 0)
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }

    fun reverse(array: DoubleArray?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = Math.max(startIndexInclusive, 0)
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }

    fun reverse(array: FloatArray?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = Math.max(startIndexInclusive, 0)
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }

    fun reverse(array: IntArray?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = Math.max(startIndexInclusive, 0)
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }

    fun reverse(array: LongArray?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = Math.max(startIndexInclusive, 0)
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }

    fun reverse(array: Array<Any?>?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = if (startIndexInclusive < 0) 0 else startIndexInclusive
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }

    fun reverse(array: ShortArray?, startIndexInclusive: Int, endIndexExclusive: Int) {
        if (array != null) {
            var i = if (startIndexInclusive < 0) 0 else startIndexInclusive
            var j = Math.min(array.size, endIndexExclusive) - 1
            while (j > i) {
                val tmp = array[j]
                array[j] = array[i]
                array[i] = tmp
                --j
                ++i
            }
        }
    }
}